exercises/chapter6/test/no-peeking/Solutions.purs

```diff
@@ -113,17 +111,11 @@ newtype Self m
 instance actionSelf :: Monoid m => Action m (Self m) where
```

Comment:
Why is the `Monoid` constraint required here? It seems like this superclass relationship is already specified by the `Action` class and that it should be possible to write it like this instead:
```purs
instance actionSelf :: Action m (Self m) where
```
Note that this question should be removed from the comments in the above lines.

Reply:
Answer:
> it's _required_ for the superclass relationship, it's not _provided_ by the superclass relationship